### ISSUE TITLE
USB redirect still broken in virt-viewer

### ISSUE THREAD

**Issue by d4g** [NixOS/nixpkgs#39596]
## Issue description
USB redirection in virt-viewer does not work.

### Steps to reproduce
- start virt-viewer
- add a usb device
- read the new error message
![image](https://user-images.githubusercontent.com/723199/39352316-a5a0ed84-4a04-11e8-8f57-f77ef379a536.png)

### NixOS details
 - system: `"x86_64-linux"`
 - host os: `Linux 4.15.16, NixOS, 18.09pre135256.6c064e6b1f3 (Jellyfish)`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.0`
 - channels(root): `"nixos-18.09pre135256.6c064e6b1f3"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos/nixpkgs`
 - nix-channel: `https://nixos.org/channels/nixos-unstable`

### Device details
 - device: `20J6S0DM00 ThinkPad T470p`
 - cpu: `Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz`
 - gpu: `Intel Corporation Device 591b (rev 04)`

### Relevant configuration
```nix
security.wrappers.spice-client-glib-usb-acl-helper.source = "${pkgs.spice-gtk}/bin/spice-client-glib-usb-acl-helper";
```

**Comment by d4g** [comment]
@xeji for me the fix of #36533 just changed the error, but did not fix the problem.
What do I need to do to change the setting to not do any fance PolicyKit stuff again? Or how do I get it to work? This package - as it currently is - does not work out of the box.

**Comment by xeji** [comment]
Looks like the polkit rule is not installed. It should work if you explicitly add  `spice-gtk` to your `environment.systemPackages`. I know this needs better documentation and should become a nixos module eventually so you can just enable everything in a single setting.

**Comment by xeji** [comment]
If you don't want to use polkit, you need to rebuild the package and follow the instructions in the comments [here](https://github.com/NixOS/nixpkgs/blob/136403655c15b830fff4f58fd90b3e1478c7ba30/pkgs/development/libraries/spice-gtk/default.nix#L17). But you still would have to put `spice-gtk` into your `environment.ststemPackages`.

**Comment by jtojnar** [comment]
For gnome-boxes, we added it to `propagetedUserEnvPkgs` https://github.com/NixOS/nixpkgs/pull/39133

**Comment by d4g** [comment]
@xeji In nixos dependency management should be handled by the packages not by the user.
If your "fix" requires the user to manually add another package, then please make this version an optional override  and make the one without policy kit default - as it was before.
Manually managing if a user should have access rights to USB is ok for me, but manually managing dependencies is not.

I got it working by disabling polkit. The version with polkit does not work by its own.

```nix
nixpkgs.config.packageOverrides = pkgs: with pkgs; {
      spice-gtk = spice-gtk.override {
        withPolkit = false;
      };
    };
```

**Comment by d4g** [comment]
The difference is that the dependency management should be part of the package management. 
Management for access rights is something differrent and should not be package dependent.

**Comment by jtojnar** [comment]
Well, the dependency is there, you just need to enable the polkit policy. You can do that by adding the polkit file to the `/var/run/current-system/sw/share/polkit-1/actions`, which is co-incidentally a side-effect of adding `spice-gtk` to `environment.systemPackages`. We could create `services.polkit.enabledActions` as an alias of `environment.systemPackages` if you would like more semantically correct option.

**Comment by xeji** [comment]
This actually is a bug, on current master the output contains no policy files at all. 

@jtojnar It looks like 82f626702314928c9c8f4ea309430e3aa0680d57 broke it by: 
```
-  PKG_CONFIG_POLKIT_GOBJECT_1_POLICYDIR = "${placeholder "out"}/share/polkit-1/actions";
+  PKG_CONFIG_POLKIT_GOBJECT_1_POLICYDIR = "share/polkit-1/actions";
```
And I just  confirmed it is fixed again as part of  #39133 by
```
-  PKG_CONFIG_POLKIT_GOBJECT_1_POLICYDIR = "share/polkit-1/actions";
+  PKG_CONFIG_POLKIT_GOBJECT_1_POLICYDIR = "$(out)/share/polkit-1/actions";
```
With that change the policy file appears again.

**Comment by xeji** [comment]
Since the fix has been merged to staging already I think we can close this issue.
@d4g sorry for the confusion.

**Comment by jtojnar** [comment]
We might want to add `spice-gtk` to propagateUserEnvPkgs` as a hack to enable the policies automatically.

**Comment by xeji** [comment]
Does that make a difference when `spice-gtk` is installed via global user config (which you need to do anyway for the setuid wrapper) ? I'm not sure how `propagateUserEnvPkgs` works.

**Comment by jtojnar** [comment]
It would be propagated to the profile you installed the package – if you added a package to `/run/current-system/sw` using `environment.systemPackages`, it would appear there as well. Obviously it does not help when running from other profiles (and it should not, for security reasons) but it works well for the system profile.

**Comment by xeji** [comment]
Thanks, I'll give that a try. 
Also I plan to create a nixos module for spice usb redirection so users can just do everything in one config option.

**Comment by d4g** [comment]
@xeji thanks for the quick response to this issue. I also appreciate adding the config option.

**Comment by xeji** [comment]
I opened #39618 for the nixos module. Closing this as fixed by #39133 .
